gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/medication/MedicationRequestMapper.java

```diff
@@ -136,16 +136,21 @@ private List<Medication> mapMedications(RCMRMT030101UKMedicationStatement medica
     }
 
     private List<MedicationRequest> mapMedicationRequestsOrder(RCMRMT030101UKEhrExtract ehrExtract,
+                                                               RCMRMT030101UKEhrComposition ehrComposition,
                                                                RCMRMT030101UKMedicationStatement medicationStatement,
                                                                String practiseCode) {
 
         return medicationStatement.getComponent()
             .stream()
             .filter(RCMRMT030101UKComponent2::hasEhrSupplyPrescribe)
             .map(RCMRMT030101UKComponent2::getEhrSupplyPrescribe)
-            .map(supplyPrescribe -> medicationRequestOrderMapper.mapToOrderMedicationRequest(ehrExtract, medicationStatement,
-                supplyPrescribe, practiseCode))
-            .filter(Objects::nonNull)
+            .map(supplyPrescribe -> medicationRequestOrderMapper.mapToOrderMedicationRequest(
```

Comment:
Just thinking... if mapToOrderMedicationRequest method inside already working with optional values then maybe it would be good to return an optional value which would allow to eliminate nulls and possibility of getting NPE...

Reply:
Seems a bit out of scope of this PR, but nothing stopping you refactoring in a separate PR.

Reply:
Will address in another PR 😄